jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResourceFactory.java

```diff
@@ -13,19 +13,63 @@
 
 package org.eclipse.jetty.util.resource;
 
+import java.io.IOException;
 import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystemNotFoundException;
+import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.util.Collections;
 
 public class PathResourceFactory implements ResourceFactory
 {
+    static final boolean ENABLE_GRAALVM_RESOURCE_SCHEME = (System.getProperty(
+        "org.graalvm.nativeimage.kind") != null);
+
+    static
+    {
+        if (ENABLE_GRAALVM_RESOURCE_SCHEME)
+        {
+            // initialize NativeImageResourceFileSystem, if necessary
+            URI uri;
+            try
+            {
+                uri = new URI("resource:/");
+                try
+                {
+                    Path.of(uri);
+                }
+                catch (FileSystemNotFoundException e)
+                {
+                    FileSystems.newFileSystem(uri, Collections.emptyMap());
+                }
+            }
+            catch (IOException | URISyntaxException | RuntimeException ignore)
+            {
+              // ignore
+            }
```

Comment:
I don't like such specific initializer code in a fairly generic class.   This would probably be better put in the constructor of a `ResourceFactory` that is installed if needed.  This would also allow parameters to be set and passed in if needed.

Note that the style of initialization could probably be generalized so that any JVM supported filesystem that needs to be created can be.  i.e. it does not need to have "resource" hard coded.

Reply:
We might be able to move that initializer to all to a wrapper for `Path.of`/`Paths.get` that always tries to initialize the filesystem for the given URI upon `FileSystemNotFoundException`.

One minor issue is that I haven't found a way to convince Github's CodeQL checker to not throw crazy errors at me when using a "user-supplied" parameter to `Path.of`.